web/gridt/tests/unit/test_schemas.py

```diff
@@ -59,3 +67,80 @@ def test_movement_schema_lengths(self):
                 "name": ["Length must be between 4 and 50."],
             },
         )
+
+    def test_request_email_change_schema_invalid(self):
+        with self.app_context():
+            user = self.create_user()
+            bad_request = {
+                "password": "bad_password",
+                "new_email": "bad_email",
+            }
+            schema = RequestEmailChangeSchema()
+            schema.context = {"user": user}
+            with self.assertRaises(ValidationError) as error:
+                schema.load(bad_request)
+
+            self.assertEqual(
+                error.exception.messages,
+                {
+                    "password": ["Failed to identify user with given password."],
+                    "new_email": ["Not a valid email address."],
+                },
+            )
```

Comment:
It is good to have this in the code block of the with statement, as that is where it is defined.